pkg/gpu/nvidia/manager.go

```diff
@@ -112,10 +164,10 @@ func (ngm *nvidiaGPUManager) ListDevices() map[string]pluginapi.Device {
 
 	switch {
 	// We will have MPS solution in the future, which will be mutually exclusive with the time-sharing solution.
-	case ngm.gpuConfig.MaxTimeSharedClientsPerGPU > 0:
+	case ngm.gpuConfig.GPUSharingConfig.MaxSharedClientsPerGPU > 0:
```

Comment:
Does this change only mean to change the time-sharing to use the new parameter, or this change mean to expand the usage to time-sharing + MPS?

If it is the former one, it missed to check the sharing strategy is time-sharing.
If it is the latter one:
1). how will the virtual device ID from MPS be in the same form as time-sharing (we need to configure related code for the MPS specific initContainer in driver installer?)
2). we'll need to update the comments reflect it includes MPS, and update the package name `timesharing.VirtualToPhysicalDeviceID` in follow-up PRs.

Reply:
It's the latter.  Basically MPS is a process running on the host, that has exclusive control of the GPUs. However MPS is transparent to any CUDA application (i.e. application doesn't need to know whether MPS is running or not). The only thing necessary for containers to consume the GPU through MPS is the directory used by MPS for named pipes and unix sockets. This additional directory is also mounted by the device plugin, but the rest of the device discovery and consumption remains unchanged.

Agree that we need to change the timesharing package name, will update that.